chia/rpc/wallet_rpc_api.py

```diff
@@ -1539,13 +1552,27 @@ async def did_transfer_did(self, request) -> EndpointResult:
         }
 
     async def did_sign_message(self, request) -> EndpointResult:
-        wallet_id = uint32(request["wallet_id"])
-        did_wallet: WalletProtocol = self.service.wallet_state_manager.wallets[wallet_id]
+        """
+        Given a DID ID, sign the message by the DID's private key.
+        :param request:
+        :return:
+        """
+        if request["did_id"].startswith("did"):
```

Comment:
It would be nice to use the `AddressType` utils to avoid hard-coding bech32m prefixes.

```suggestion
        if is_valid_address(request["did_id"], {AddressType.DID), self.service.config):
```